djangocms_googlemap/models.py

```diff
@@ -1,90 +1,287 @@
+    )
+    lat = models.FloatField(
+        verbose_name=_('Latitude (lat)'),
+        null=True,
+        blank=True,
+        help_text=_('Default GPS latitute (e.g. "46.947973").'),
+    )
+    lng = models.FloatField(
+        verbose_name=_('Longitude (lng)'),
+        null=True,
+        blank=True,
+        help_text=_('Default GPS longitude (e.g. "7.447446").'),
+    )
     zoom = models.PositiveSmallIntegerField(
-        _("zoom level"), choices=ZOOM_LEVELS, default=13)
-
-    lat = models.DecimalField(
-        _('latitude'), max_digits=10, decimal_places=6, null=True, blank=True,
-        help_text=_('Use latitude & longitude to fine tune the map position.'))
+        verbose_name=_('Zoom level'),
+        choices=ZOOM_LEVEL_CHOICES,
+        default=13,
+    )
 
-    lng = models.DecimalField(
-        _('longitude'), max_digits=10, decimal_places=6, null=True, blank=True)
-
-    route_planer_title = models.CharField(
-        _("route planner title"), max_length=150, blank=True, null=True,
-        default=_('Calculate your fastest way to here'))
-
-    route_planer = models.BooleanField(_("route planner"), default=False)
+    # control settings
+    map_type_control = models.CharField(
+        verbose_name=_('Map type control'),
+        choices=MAP_TYPE_CHOICES,
+        default=MAP_TYPE_CHOICES[0][0],
+        max_length=255,
+    )
+    zoom_control = models.BooleanField(
+        verbose_name=_('Zoom control'),
+        default=True,
+        help_text=_('Show the Zoom control on the map.'),
+    )
+    street_view_control = models.BooleanField(
+        verbose_name=_('Street view control'),
+        default=True,
+        help_text=_('Show the Street View Pegman control on the map.'),
+    )
+    rotate_control = models.BooleanField(
+        verbose_name=_('Rotate control'),
+        default=True,
+        help_text=_('Show the Rotate control on the map.'),
+    )
+    scale_control = models.BooleanField(
+        verbose_name=_('Scale control'),
+        default=True,
+        help_text=_('Show the Scale control on the map.'),
+    )
+    fullscreen_control = models.BooleanField(
+        verbose_name=_('Fullscreen control'),
+        default=True,
+        help_text=_('Show Fullscreen control on the map.'),
+    )
+    pan_control = models.BooleanField(
+        verbose_name=_('Pan control'),
+        default=True,
+        help_text=_('Show the Pan control on the map.'),
+    )
+    double_click_zoom = models.BooleanField(
+        verbose_name=_('Double click to zoom'),
+        default=True,
+        help_text=_('Enable zoom and center on double click on the map.'),
+    )
+    draggable = models.BooleanField(
+        verbose_name=_('Draggable map'),
+        default=True,
+        help_text=_('Enable mouse dragging on the map.'),
+    )
+    keyboard_shortcuts = models.BooleanField(
+        verbose_name=_('Keyboard shortcuts'),
+        default=True,
+        help_text=_('Enable map control via the keyboard.'),
```

Comment:
```
verbose_name=_('Enable keyboard shortcuts'),
help_text=_('Allows the user to control the map using the keyboard.'),
```

How is different from ``pan_control`` above? I can't even find some of these options at https://developers.google.com/maps/documentation/javascript/controls!